### PR TITLE
Fix/chat-390-01: Make sure that deleted or disabled users can not be added into chat-room

### PR DESCRIPTION
**Pull request by mdiha**
When creating a new chat room, we find in the suggested list of users, users who are deleted or disabled.
*Fix:
In order to fix this issue i added an action listener for deleting users in chat's database at the same time they are deleted from the platform's database.
Same For the disabled users.
*Migration of existing users:
To migrate existing users we avoided adding an upgrade plugin due to chat architecture, and as an alternative we made sure that the existing users gets updated by several manners:
-at login the user gets updated if not already updated.
-by clicking on a contact from chat contact list, the selected contact gets updated by the time. For team rooms or space rooms that are selected the participants of all the room gets updated if they are not already updated.
-by adding a new room the users which appears in the suggested list gets updated by the time.

### CODE REVIEW

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java [hunk 1]
```diff
@@ -359,6 +362,8 @@ public void addUserEmail(String user, String email)
       doc.put("_id", user);
       doc.put("user", user);
       doc.put("email", email);
+      doc.put("isEnabled", Boolean.TRUE.toString());
```
Comment (ahamdi) [inline server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java]: No need to update the user status when adding his email.

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java [hunk 2]
```diff
@@ -335,6 +336,8 @@ public void addUserFullName(String user, String fullname)
       doc.put("_id", user);
       doc.put("user", user);
       doc.put("fullname", fullname);
+      doc.put("isEnabled", Boolean.TRUE.toString());
```
Comment (ahamdi) [inline server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java]: No need to update the user status when adding his fullName

File: application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue [hunk 3]
```diff
@@ -74,7 +74,7 @@ export default {
     }
   },
   updated() {
-    if (this.mq === 'desktop') { // set autofocus only on desktop
+    if (this.mq === 'desktop' && (this.contact.isEnabledUser === 'true' || this.contact.isEnabledUser === 'null')) { // set autofocus only on desktop
```
[the remark this answers is not in the document]
Comment (mdiha) [reply application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue]: Okay

File: application/src/main/webapp/vue-app/components/ExoChatRoomParticipants.vue [hunk 4]
```diff
@@ -164,18 +164,42 @@ export default {
...
+            usersToUpdate = data.users.filter( user => 
+              user.isEnabled === 'null'
+            );
+            if(usersToUpdate.length > 0) {
+              usersToUpdate.forEach( finalContact => {
+                chatServices.getUserState(finalContact.name).then(userState => {
+                  chatServices.updateUser(eXo.chat.userSettings, finalContact.name, userState.isDeleted, userState.isEnabled);
+                });
+              });
+            } else {
```
[the remark this answers is not in the document]
Comment (mdiha) [reply application/src/main/webapp/vue-app/components/ExoChatRoomParticipants.vue]: Okay will fix it

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java (hunk 1 above)
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java]: Resolved

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java (hunk 2 above)
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/UserMongoDataStorage.java]: Resolved

File: application/src/main/webapp/vue-app/components/modal/ExoChatRoomFormModal.vue [hunk 5]
```diff
@@ -138,14 +138,31 @@ export default {
...
+              chatServices.getUserState(finalContact.name).then(userState => {
+                chatServices.updateUser(eXo.chat.userSettings, finalContact.name, userState.isDeleted, userState.isEnabled);
+              });
+            });
+          } else {
+            callback(data.users.filter(user => user.name !== eXo.chat.userSettings.username));
+          }
         }
       });
+      if(usersToUpdate.length > 0) {
```
[the remark this answers is not in the document]
Comment (mdiha) [reply application/src/main/webapp/vue-app/components/modal/ExoChatRoomFormModal.vue]: Fixed.

File: server-embedded/src/main/java/org/exoplatform/chat/services/ChatServiceImpl.java [hunk 6]
```diff
@@ -144,7 +145,10 @@ public void delete(String room, String sender, String messageId)
         UserBean userBean = userService.getUser(room);
         String username = userBean.getName();
         if (username == null) {
-          List<UserBean> users = userService.getUsersInRoomChatOneToOne(room);
+          List<UserBean> users = userService.getUsersInRoomChatOneToOne(room)
```
[the remark this answers is not in the document]
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/services/ChatServiceImpl.java]: Resolved

File: server-embedded/src/main/java/org/exoplatform/chat/services/ChatServiceImpl.java [hunk 7]
```diff
@@ -345,7 +349,10 @@ private boolean isMemberOfRoom(String username, String roomId) {
       roomMembers = userService.getUsersFilterBy(null, roomId, ChatService.TYPE_ROOM_SPACE);
     } else {
       roomMembers = new ArrayList<>();
-      List<UserBean> userBeans = userService.getUsersInRoomChatOneToOne(roomId);
+      List<UserBean> userBeans = userService.getUsersInRoomChatOneToOne(roomId)
```
[the remark this answers is not in the document]
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/services/ChatServiceImpl.java]: Resolved

**Pull request merged**